LordMicrostrainIMU's update method differs in signature

## 🐛 Bug Report

LordMicrostrainIMU's `update` method differs from IMUBase. `update` method is designed to return None, `data_packets` can be set as a class attribute that the user can access after update is called.

```
    def update(self, timeout: int = 500, max_packets: int = 1, return_packets: bool = False) -> Optional[list[Any]]:
        """
        Get IMU data from the Lord Microstrain IMU
        """
        data_packets = self._node.getDataPackets(timeout=timeout, maxPackets=max_packets)
        data_points = data_packets[-1].data()
        self._data = {data.channelName(): data.as_float() for data in data_points}

        if return_packets:
            return data_packets
```

Problem:
```
imu.py:110: error: Missing return statement  [return]
imu.py:110: error: Signature of "update" incompatible with supertype "SensorBase"  [override]
imu.py:110: note:      Superclass:
imu.py:110: note:          def update(self) -> None
imu.py:110: note:      Subclass:
imu.py:110: note:          def update(self, timeout: int = ..., max_packets: int = ..., return_packets: bool = ...) -> list[Any] | None
```

--- a/opensourceleg/sensors/imu.py
+++ b/opensourceleg/sensors/imu.py
@@ -70,6 +70,9 @@ class LordMicrostrainIMU(IMUBase):
         port: str = "/dev/ttyUSB0",
         baud_rate: int = 921600,
         frequency: int = 200,
+        update_timeout: int = 500,
+        max_packets: int = 1,
+        return_packets: bool = False,
     ) -> None:
         """
         Initialize the LordMicrostrainIMU sensor.
@@ -78,10 +81,16 @@ class LordMicrostrainIMU(IMUBase):
             port (str, optional): Serial port for the IMU. Defaults to "/dev/ttyUSB0".
             baud_rate (int, optional): Baud rate for the serial connection. Defaults to 921600.
             frequency (int, optional): Data streaming frequency in Hz. Defaults to 200.
+            update_timeout (int, optional): Timeout for data packet retrieval in milliseconds. Defaults to 500.
+            max_packets (int, optional): Maximum number of data packets to retrieve. Defaults to 1.
+            return_packets (bool, optional): If True, returns the raw data packets. Defaults to False.
         """
         self._port = port
         self._baud_rate = baud_rate
         self._frequency = frequency
+        self._update_timeout = update_timeout
+        self._max_packets = max_packets
+        self._return_packets = return_packets
         self._is_streaming = False
         self._connection = None
         self._data: dict[str, float] = {}
@@ -170,7 +179,7 @@ class LordMicrostrainIMU(IMUBase):
         else:
             LOGGER.error(f"Failed to ping the IMU at {self.port}")
 
-    def update(self, timeout: int = 500, max_packets: int = 1, return_packets: bool = False) -> Union[None, Any]:
+    def update(self) -> Union[None, Any]:
         """
         Retrieve and update IMU data from the sensor.
 
@@ -182,11 +191,11 @@ class LordMicrostrainIMU(IMUBase):
         Returns:
             Union[None, Any]: Returns the data packets if `return_packets` is True; otherwise, None.
         """
-        data_packets = self._node.getDataPackets(timeout=timeout, maxPackets=max_packets)
+        data_packets = self._node.getDataPackets(timeout=self._update_timeout, maxPackets=self._max_packets)
         data_points = data_packets[-1].data()
         self._data = {data.channelName(): data.as_float() for data in data_points}
 
-        if return_packets:
+        if self._return_packets:
             return data_packets
         else:
             return None

--- a/opensourceleg/sensors/imu.py
+++ b/opensourceleg/sensors/imu.py
@@ -136,6 +136,24 @@ class LordMicrostrainIMU(IMUBase):
 
         return channels
 
+    def set_update_timeout(self, timeout: int) -> None:
+        """
+        Set the update timeout for the sensor.
+        """
+        self._update_timeout = timeout
+
+    def set_max_packets(self, max_packets: int) -> None:
+        """
+        Set the maximum number of packets to retrieve.
+        """
+        self._max_packets = max_packets
+
+    def set_return_packets(self, return_packets: bool) -> None:
+        """
+        Set the return packets flag.
+        """
+        self._return_packets = return_packets
+
     def start(self) -> None:
         """
         Start the Lord Microstrain IMU sensor.
@@ -191,11 +209,11 @@ class LordMicrostrainIMU(IMUBase):
         Returns:
             Union[None, Any]: Returns the data packets if `return_packets` is True; otherwise, None.
         """
-        data_packets = self._node.getDataPackets(timeout=self._update_timeout, maxPackets=self._max_packets)
+        data_packets = self._node.getDataPackets(timeout=self.update_timeout, maxPackets=self.max_packets)
         data_points = data_packets[-1].data()
         self._data = {data.channelName(): data.as_float() for data in data_points}
 
-        if self._return_packets:
+        if self.return_packets:
             return data_packets
         else:
             return None
@@ -248,6 +266,27 @@ class LordMicrostrainIMU(IMUBase):
             bool: True if streaming; otherwise, False.
         """
         return self._is_streaming
+
+    @property
+    def update_timeout(self) -> int:
+        """
+        Get the update timeout for the sensor.
+        """
+        return self._update_timeout
+
+    @property
+    def max_packets(self) -> int:
+        """
+        Get the maximum number of packets to retrieve.
+        """
+        return self._max_packets
+
+    @property
+    def return_packets(self) -> bool:
+        """
+        Get the return packets flag.
+        """
+        return self._return_packets
 
     @property
     def data(self) -> dict[str, float]:

--- a/opensourceleg/sensors/imu.py
+++ b/opensourceleg/sensors/imu.py
@@ -199,12 +199,8 @@ class LordMicrostrainIMU(IMUBase):
 
     def update(self) -> Union[None, Any]:
         """
-        Retrieve and update IMU data from the sensor.
-
-        Args:
-            timeout (int, optional): Timeout for data packet retrieval in milliseconds. Defaults to 500.
-            max_packets (int, optional): Maximum number of data packets to retrieve. Defaults to 1.
-            return_packets (bool, optional): If True, returns the raw data packets. Defaults to False.
+        Retrieve and update IMU data from the sensor. To modify update parameters, use the set_update_timeout,
+        set_max_packets, and set_return_packets methods.
 
         Returns:
             Union[None, Any]: Returns the data packets if `return_packets` is True; otherwise, None.

--- a/tests/test_sensors/test_imu.py
+++ b/tests/test_sensors/test_imu.py
@@ -245,12 +245,16 @@ def test_ping_success(sample_imu: MockLordMicrostrainIMU):
 
 # Test update
 def test_update(sample_imu: MockLordMicrostrainIMU):
+    sample_imu.set_update_timeout(400)
+    sample_imu.set_max_packets(3)
+    sample_imu.set_return_packets(True)
+
     sample_imu.start()
     assert sample_imu._data == {}
-    sample_imu.update(500, 2, False)
+    sample_imu.update()
     assert sample_imu._data == {"mockdata": 20}
 
-    return_val = sample_imu.update(400, 3, True)
+    return_val = sample_imu.update()
     assert len(return_val) == 3
     for i in range(0, 3):
         assert isinstance(return_val[i], MockDataPockets)

